cmd/kubectl-gadget/utils/trace.go

```diff
@@ -290,32 +455,158 @@ RetryLoop:
 		// Don't print warnings if at least one node succeeded. This avoids showing
 		// warnings together with the actual output generated by other nodes.
 		if successNodeCount == 0 {
-			printTraceFeedback(contextLogger.Warningf, nodeWarnings)
+			printWarningFunction := func(format string, args ...interface{}) { fmt.Printf(format, args) }
+
+			printTraceFeedback(printWarningFunction, nodeWarnings)
 		}
 
+		printErrorFunction := func(format string, args ...interface{}) { fmt.Fprintf(os.Stderr, format, args) }
+
 		// Print errors even if other nodes succeeded.
-		printTraceFeedback(contextLogger.Errorf, nodeErrors)
+		printTraceFeedback(printErrorFunction, nodeErrors)
 
 		if successNodeCount == 0 {
-			deleteTraces(contextLogger, traceRestClient, traceID)
-			contextLogger.Fatalf("Failed to run the gadget on all nodes: None of them succeeded")
+			return errors.New("Failed to run the gadget on all nodes: None of them succeeded")
 		}
+
 		break RetryLoop
 	}
 
+	client, err := k8sutil.NewClientsetFromConfigFlags(KubernetesConfigFlags)
+	if err != nil {
+		return fmt.Errorf("Error in creating setting up Kubernetes client: %w", err)
+	}
+
 	if customResultsDisplay == nil {
-		if outputMode != "Stream" {
-			panic(fmt.Errorf("OutputMode=%q needs a custom display function", outputMode))
+		/*
+		 * If we arrived here, it means we have at least one successful node, so it
+		 * is safe to use trace pointer.
+		 */
+		if trace.Spec.OutputMode != "Stream" {
+			panic(fmt.Errorf("OutputMode=%q needs a custom display function", trace.Spec.OutputMode))
 		}
-		genericStreamsDisplay(contextLogger, client, params, &results, transformLine)
+
+		params := reconstructParamsFromTrace(trace)
+		params.Verbose = verbose
+
+		genericStreamsDisplay(client, params, &traces, transformLine)
 	} else {
-		customResultsDisplay(contextLogger, nodes, &results)
+		nodes, err := client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return fmt.Errorf("Error in listing nodes: %w", err)
+		}
+
+		customResultsDisplay(nodes, &traces)
 	}
-	deleteTraces(contextLogger, traceRestClient, traceID)
+
+	return nil
+}
+
+// DeleteTrace deletes the traces whom ID is given as parameter using
```

Comment:
Traces aren't people.
```suggestion
// DeleteTrace deletes the traces for the given traceID using
```